src/main.cpp

```diff
@@ -4559,28 +4562,63 @@ bool ReceivedBlockTransactions(
         deque<CBlockIndex*> queue;
         queue.push_back(pindexNew);
 
+        CCoinsViewCache view(pcoinsTip);
```

Comment:
After a bunch of discussion, we concluded that this can become inaccurate; instead, all of this needs to be done in ConnectBlock where we know we have a valid view.